core/src/test/java/feign/RetryerTest.java

```diff
@@ -74,4 +77,23 @@ protected long currentTimeMillis() {
   public void neverRetryAlwaysPropagates() {
     Retryer.NEVER_RETRY.continueOrPropagate(new RetryableException(null, null, new Date(5000)));
   }
+
+  @Test
+  public void defaultRetryerShouldPropagateFeignExceptionWhenThreadGetsInterrupted() {
+    Retryer retryer = new Retryer.Default();
+    Thread thread = new Thread(() ->
+            assertThatThrowsExceptionAndThreadIsInterrupted(RetryableException.class,
+                    () -> retryer.continueOrPropagate(new RetryableException(null, null))));
+    thread.start();
+    thread.interrupt();
```

Comment:
How can this test fail with no assertions while swallowing all exceptions in another thread?